core/src/main/java/amino/run/oms/migrationdecision/MetricWatcher.java

```diff
@@ -0,0 +1,282 @@
+        } catch (MicroServiceNotFoundException e) {
+            logger.warning(
+                    String.format(
+                            "Failed to migrate replica [%s] as micro service not available",
+                            replicaID));
+            e.printStackTrace();
+            return false;
+        } catch (RemoteException e) {
+            logger.warning(
+                    String.format(
+                            "Failed to migrate replica [%s] with remote exception : %s",
+                            replicaID, e.getMessage()));
+            e.printStackTrace();
+            return false;
+        }
+
+        return true;
+    }
+
+    private long aggregatedProcessingTime(ReplicaID replicaID)
+            throws MicroServiceNotFoundException, MicroServiceReplicaNotFoundException,
+                    KernelServerNotFoundException {
+        /*
+           To measure time consumption by vertex Vk we use the following formula
+           tk = Tk,posk+i + summation { (Ski+Sik)/Rposk,posi + Lposk,posi }
+        */
+
+        // get current kernel server for replica
+        InetSocketAddress replicaKernelServer =
+                microServiceManager.getReplicaDispatcher(replicaID).getHost();
+        // get metric stat for current replica kernel server
+        Map<InetSocketAddress, NodeMetric> replicaNodeMetric =
+                serverManager.getKernelServerMetric(replicaKernelServer);
+
+        // get metrics stat of RPC processing time from different kernel client
+        // TODO currently considering processing time from different kernel client for simplicity.
+        //  Need to get processing time within micro service replica
+        Map<InetSocketAddress, RPCMetric> replicaMetrics =
+                microServiceManager.getMicroServiceMetric(replicaID);
+
+        // calculate time factor impact on RPC with latency and bandwidth
+        long averageExecutionTime = 0;
+        long summation = 0;
+        long data;
+        NodeMetric nodeMetric;
+        for (Map.Entry<InetSocketAddress, RPCMetric> replicaMetric : replicaMetrics.entrySet()) {
+            averageExecutionTime += replicaMetric.getValue().elapsedTime;
+
+            data = replicaMetric.getValue().dataSize;
+            nodeMetric = replicaNodeMetric.get(replicaMetric.getKey());
+            // kernel client utilized by micro service but not running in kernel server are
+            // ignored in process time computation because no metric data available of kernel server
+            // from there.
+            if (nodeMetric == null) {
+                logger.info("skipping kernel client not part of kernel server");
+                continue;
+            }
+
+            summation += (data / nodeMetric.rate + nodeMetric.latency);
+        }
+
+        if (!replicaMetrics.isEmpty()) {
+            averageExecutionTime = averageExecutionTime / replicaMetrics.size();
+        }
+
+        return averageExecutionTime + summation;
```

Comment:
It's not clear to me that you can simply add the averageExecutionTime (which is actually the average elapsed time for RPC's) to the summation of all data transfer times and get a sensible result.  Surely you should either add an average to an average, or a summation to a summation?